File: c_third_lesson/src/service_node.cpp
#include "ros/ros.h"
#include "c_third_lesson/Service.h"

bool add(
    c_third_lesson::Service::Request &request, 
    c_third_lesson::Service::Response &response
    )
{
    response.sum = request.a + request.b;
    ROS_INFO("request: x=%ld, y=%ld", (long int)request.a, (long int)request.b);
    ROS_INFO("sending back response: [%ld]", (long int)response.sum);
    return true;
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "service_node");

    ros::NodeHandle nh;

    ros::ServiceServer service = nh.advertiseService("add_two_ints", add);
    ROS_INFO("Ready to add two ints.");
    ros::spin();

    return 0;
}

File: c_third_lesson/src/service_node.py
#! /usr/bin/python3

import rospy
from c_third_lesson.srv import Service, ServiceResponse, ServiceRequest


def add_callback(req: ServiceRequest):
    print("Returning [%s + %s = %s]" % (req.a, req.b, (req.a + req.b)))
    return ServiceResponse(req.a + req.b)


def add():
    rospy.init_node('add_two_ints')
    service = rospy.Service('add_two_ints', Service, add_callback)
    print("Ready to add two ints.")
    rospy.spin()

if __name__ == "__main__":
    add()